### PR TITLE
Add support for per-service root url definitions

### PR DESCRIPTION
**Pull request by ircmaxell**
This adds support for per-service root url definitions.

### CODE REVIEW

File: src/Google/Service/Resource.php [hunk 1]
```diff
@@ -58,6 +61,7 @@ class Google_Service_Resource
 
   public function __construct($service, $serviceName, $resourceName, $resource)
   {
+  	$this->rootUrl = $service->rootUrl;
```
Comment (ianbarber) [inline src/Google/Service/Resource.php]: Fix tab.

File: tests/general/ResourceTest.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    $this->client->expects($this->any())
+          ->method("getAuth")
+          ->will($this->returnValue(new Google_Auth_Simple($this->client)));
+    $this->service = new Test_Google_Service($this->client);
+  }
+
+  public function testCallFailure()
+  {
+    $resource = new Google_Service_Resource($this->service, "test", "testResource", "testMethod");
+    $this->setExpectedException("Google_Exception", "Unknown function: test->testResource->someothermethod()");
```
Comment (ianbarber) [inline tests/general/ResourceTest.php]: line length - can we keep to 100 width.

File: src/Google/Service/Resource.php [hunk 3]
```diff
@@ -193,7 +197,12 @@ public function call($name, $arguments, $expected_class = null)
         null,
         $postBody
     );
-    $httpRequest->setBaseComponent($this->client->getBasePath());
+
+    if ($this->rootUrl) {
```
Comment (ianbarber) [inline src/Google/Service/Resource.php]: I think we'll need to bump to 1.2 if we're being semver-ish, as this is potentially a BC break - in the past we would also use the configured basepath if one, but now we will override a configured base path with the one out of the service. Realistically I'm not sure this is really going to hit anyone though - WDYT? We should bump the version number in Client.php regardless to something.

File: tests/general/ResourceTest.php [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  {
+    parent::__construct($client);
+    $this->rootUrl = "";
+    $this->servicePath = "";
+    $this->version = "v1beta1";
+    $this->serviceName = "test";
+  }
+}
+
+class ResourceTest extends PHPUnit_Framework_TestCase
```
Comment (ianbarber) [inline tests/general/ResourceTest.php]: Nice test!

**Comment by ianbarber** [comment]
For future people: anthony is a Googler, ignore clabot.

**Comment by ianbarber** [comment]
Also: Test seems to be failing.

File: tests/general/ResourceTest.php (hunk 4 above)
Comment (ircmaxell) [inline tests/general/ResourceTest.php]: Thanks!

File: src/Google/Service/Resource.php (hunk 3 above)
Comment (ircmaxell) [inline src/Google/Service/Resource.php]: TBH, I could go either way. Considering that it's a completely undocumented parameter in public docs, and the only documented reference is in config.php:

      // Don't change these unless you're working against a special development
      // or testing environment.
      'base_path' => 'https://www.googleapis.com',

I could see it going either way. It's up to you.

I'll bump the Client.php version number in the PR as well.

**Pull request merged**